lib/Importer/ONNXModelLoader.cpp

```diff
@@ -208,8 +215,38 @@ llvm::Expected<Pads> getPads(const ArgumentDictionaryTy &dict) {
     if (padStr == "VALID") {
       // Return default value 0 for pads.
       return Pads({0, 0, 0, 0});
+    } else if (padStr == "SAME_UPPER" || padStr == "SAME_LOWER") {
+      unsigned_t top, left, bottom, right;
+      // From https://arxiv.org/pdf/1603.07285.pdf 2.4,
+      // o = floor((i + 2*p - k)/s) + 1
+      // Also, from https://github.com/onnx/onnx/blob/master/docs/Operators.md
+      // output_spatial_shape[i] =
+      //     ceil(input_spatial_shape[i] / strides_spatial_shape[i])
+      // pad_shape[i] =
+      //     (output_spatial_shape[i] - 1) * strides_spatial_shape[i]
+      //         + kernel_spatial_shape[i] - input_spatial_shape[i]
+      // Use the smallest padding possible out of the possible options.
+      std::vector<unsigned_t> pdim(2); // Total Paddding, HW.
```

Comment:
This probably should be small vector size 2 but the other things (sdim, kdim, etc) aren't either so up to you